spec/unit/deploy_task_spec.rb

```diff
@@ -20,11 +20,37 @@
 describe DeployTask do
   include FakeFS::SpecHelpers
 
-  let(:store) { SystemDescriptionStore.new }
   let(:deploy_task) { DeployTask.new }
   let(:system_description) {
-    SystemDescription.new("test", store)
+    create_test_description(json: <<-EOF, store: SystemDescriptionStore.new, name: "test")
+      {
+        "packages": [
+          {
+            "name": "kernel-desktop",
+            "version": "3.7.10",
+            "release": "1.0",
+            "arch": "x86_64",
+            "vendor": "SUSE LINUX Products GmbH, Nuernberg, Germany",
+            "checksum": "2a3d5b29179daa1e65e391d0a0c1442d"
+          }
+        ],
+        "repositories": [
+          {
+            "alias": "nodejs",
+            "name": "nodejs",
+            "type": "rpm-md",
+            "url": "http://d.opensuse.org/r/devel:/languages:/nodejs/openSUSE_13.1/"
+          }
+        ],
+        "os": {
+          "name": "SUSE Linux Enterprise Server 11",
+          "version": "11 SP3",
+          "architecture": "x86_64"
+        }
+      }
+    EOF
```

Comment:
I think this can be replaced by the factory, since this entries seem to be all generic. The main difference would be an openSUSE 13.1 os I suppose.
```
    create_test_description(
      scopes: ["os", "repositories", "os"]   name: "test", store: SystemDescriptionStore.new
   )
```